docs/csharp/language-reference/keywords/in-parameter-modifier.md

```diff
@@ -37,36 +37,91 @@ class CS0663_Example
 }
 ```
   
-Overloading based on the presence of `in` is allowed, but generates a compiler warning:  
+Overloading based on the presence of `in` is allowed:  
   
 ```csharp
 class InOverloads
 {
-    // Discouraged. Calling SampleMethod(value) is ambiguous.
     public void SampleMethod(in int i) { }
     public void SampleMethod(int i) { }
 }
 ```
 
-Properties or constants may be passed as `in` parameters, because the calling method may not modify their values.
-  
-You can't use the `in`, `ref`, and `out` keywords for the following kinds of methods:  
-  
-- Async methods, which you define by using the [async](../../../csharp/language-reference/keywords/async.md) modifier.  
-  
-- Iterator methods, which include a [yield return](../../../csharp/language-reference/keywords/yield.md) or `yield break` statement.  
+## Overload resolution rules
+
+You can understand the overload resolution rules for methods with by value vs. `in` arguments through understanding the motivation for `in` arguments. Defining methods using `in` parameters is a potential performance optimization. Some `struct` type arguments may be large in size, and when methods are called in tight loops or critical code paths, the cost of copying those structures is critical. Methods declare `in` parameters to specify that arguments may be passed by reference safely because the called method does not modify the state of that argument. Passing those arguments by reference avoids the (potentially) expensive copy. 
+
+Specifying `in` on arguments at the call site is typically optional. There is no semantic difference between passing arguments by value and passing them by reference using the `in` modifier. The `in` modifier at the call site is optional because you don't need to indicate that the argument's value might be changed. You explicitly add the `in` modifier at the call site to ensure the argument is passed by reference, not by value. Explicitly using `in` has two effects:
+
+First, specifying `in` at the call site forces the compiler to select a method defined with a matching `in` parameter. Otherwise, when two methods differ only in the presence of `in`, the by value overload is a better match.
+
+Second, specifying `in` declares your intent to pass an argument by reference. The argument used with `in` must represent a location that can be directly referred to. The same general rules for `out` and `ref` arguments apply: You cannot use constants, ordinary properties, or other expressions that produce values. Omitting `in` at the call site informs the compiler that you will allow the compiler to create a temporary variable to pass by read-only reference to the method. The compiler creates a temporary variable to overcome several restrictions with `in` arguments:
```

Comment:
This remains confusing -- it was only on the fourth reading that I noticed the "Omitting...". I think that the issue can be addressed by adding "Otherwise, omitting...", which also makes it parallel to the previous paragraph.